diff_cover/violationsreporters/java_violations_reporter.py

```diff
@@ -0,0 +1,224 @@
+"""
+Classes for querying the information in a test coverage report.
+"""
+from __future__ import unicode_literals
+
+from collections import defaultdict
+
+import itertools
+from xml.etree import cElementTree
+from diff_cover.command_runner import run_command_for_code
+from diff_cover.git_path import GitPathTool
+from diff_cover.violationsreporters.base import BaseViolationReporter, Violation, RegexBasedDriver, QualityDriver
+
+
+class CloverXmlCoverageReporter(BaseViolationReporter):
+    """
+    Query information from a Clover XML coverage report.
+    """
+
+    def __init__(self, xml_roots):
+        """
+        Load the Clover XML coverage report represented
+        by the cElementTree with root element `xml_root`.
+        """
+        super(CloverXmlCoverageReporter, self).__init__("XML")
+        self._xml_roots = xml_roots
+
+        # Create a dict to cache violations dict results
+        # Keys are source file paths, values are output of `violations()`
+        self._info_cache = defaultdict(list)
+
+    @staticmethod
+    def _get_src_path_line_nodes(xml_document, src_path):
+        """
+        Return a list of nodes containing line information for `src_path`
+        in `xml_document`.
+
+        If file is not present in `xml_document`, return None
+        """
+        files = [file_tree
+                 for file_tree in xml_document.findall(".//file")
+                 if file_tree.get('path') == src_path
```

Comment:
@Bachmann1234 @peterg79 Before the equality check, we probably need to sanitize the path to unix path. Something like this [1]. Due to this clover reports generated under windows are not working with "GIT BASH"

[1] https://github.com/Bachmann1234/diff-cover/blob/da45062f066e94e0a65a84310b812c54cc423be9/diff_cover/violationsreporters/violations_reporter.py#L67

Reply:
@rishi05091989 Do you have some sample data I can use to test? I have no access to a windows computer these days.

Reply:
@peterg79 Thanks for your reply. The file tag in clover report looks like this,
```
<file path="C:\Users\guides_rishi\rishi_guides\guides\bundles\ds-guide-core\src\main\java\com\guide\themes\model\BreakpointInfo.java" name="BreakpointInfo.java">
```
 Also, is it possible to use the clover and cobertura XML together ? As of today, only one coverage report can be used at a time(mixed is not supported).

Reply:
> is it possible to use the clover and cobertura XML together
yes, with the latest release: https://github.com/Bachmann1234/diff-cover/pull/87

> need to sanitize the path to unix path
I'll take a look at it soon